### DISCUSSION TITLE
Why does SoftDelete uses cloned model?

### DISCUSSION THREAD

**Discussion by wurst-hans** [phalcon/cphalcon#15408] [Q&A]
Why does [SoftDelete behavior](https://github.com/phalcon/cphalcon/blob/master/phalcon/Mvc/Model/Behavior/SoftDelete.zep#L69) work with a clone of original model?
What could happen if we just call `save()` on the original model?

**Comment by Jeckerson** [comment]
Could you please specify your concerns about it? With some basic examples...

**Reply by wurst-hans** [reply]
I have no concerns. I just want to understand why this is done. In source code there is a comment, that `save()` is done on a clone of model to ensure a _clean operation_.
Has this been done, because you fear any interferences when calling `save()` on the original model while `delete()` is running currently on it? The only thing I can imagine is, that the `skipOperation()` called on original model might have side effect when executing `save()` on model instead of clone. But I have no clue.
